src/app/fdctl/run/tiles/fd_dedup.c

```diff
@@ -130,6 +132,17 @@ after_frag( void *             _ctx,
 
   fd_dedup_ctx_t * ctx = (fd_dedup_ctx_t *)_ctx;
 
+  ulong payload_sz = *opt_sz;
+  ulong txn_off = fd_ulong_align_up( payload_sz, 2UL );
+
+  /* Ensure sufficient trailing space for parsing. */
+  if( FD_UNLIKELY( txn_off>FD_TPU_DCACHE_MTU - FD_TXN_MAX_SZ - sizeof(ushort)) ) {
+    FD_LOG_ERR(( "got malformed txn (sz %lu) insufficient space left in dcache for fd_txn_t", payload_sz ));
+  }
+
+  uchar    * dcache_entry = fd_chunk_to_laddr( ctx->out_mem, ctx->out_chunk );
+  fd_txn_t * txn = (fd_txn_t *)(dcache_entry + txn_off);
```

Comment:
In both paths we expect txn at the same memory address. From verify, tx is already parsed and txn filled, from gossip it's not.